numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2735,30 +2764,24 @@ def new(
         additional_flags=None,
     ):
         driver_ver = driver.get_version()
-        if config.CUDA_ENABLE_MINOR_VERSION_COMPATIBILITY and driver_ver >= (
-            12,
-            0,
-        ):
-            raise ValueError("Use CUDA_ENABLE_PYNVJITLINK for CUDA >= 12.0 MVC")
-        if config.CUDA_ENABLE_PYNVJITLINK and driver_ver < (12, 0):
-            raise ValueError("Enabling pynvjitlink requires CUDA 12.")
-        if config.CUDA_ENABLE_PYNVJITLINK:
-            linker = PyNvJitLinker
-
-        elif config.CUDA_ENABLE_MINOR_VERSION_COMPATIBILITY:
-            linker = MVCLinker
+        if driver_ver < (12, 0):
+            if config.CUDA_ENABLE_MINOR_VERSION_COMPATIBILITY:
+                linker = MVCLinker
+            elif USE_NV_BINDING:
+                linker = _Linker
+            else:
+                linker = CtypesLinker
         else:
             if USE_NV_BINDING:
-                linker = CudaPythonLinker
+                linker = _Linker
             else:
                 linker = CtypesLinker
 
-        if linker is PyNvJitLinker:
-            return linker(max_registers, lineinfo, cc, lto, additional_flags)
-        elif additional_flags or lto:
-            raise ValueError("LTO and additional flags require PyNvJitLinker")
-        else:
-            return linker(max_registers, lineinfo, cc)
+        params = (max_registers, lineinfo, cc)
+        if linker is _Linker:
+            params = (*params, _have_nvjitlink(), additional_flags)
```

Comment:
I think that this is slightly off. `_LinkerBase.new` shouldn't drop the `lto` that's passed in. _But_ numba probably wants to turn `lto` on whenever it's able to, meaning it needs to to decide it that's possible at some level. @gmarkall where do you think we should set `lto=_have_nvjitlink if lto is None else lto`? Should it be in `codegen.py`? At the decorator level? Somewhere else?

Reply:
Okay that makes sense. Shall we merge 310 then and then switch it in this PR to use `_have_nvjitlink()` rather than the config variable?

Reply:
I think merging 310 and switching it to use `_have_nvjitlink()` in this PR sounds good. Following that, we still should ensure we're not dropping the `lto` in `_LinkerBase.new` - in conjunction with the observation at https://github.com/NVIDIA/numba-cuda/pull/133/files#r2167132125 (that codegen.py is calling `_Linker.new()` when the method is in `_LinkerBase`), it suggests the abstraction / design here is a little bit mixed up between `_Linker` and `_LinkerBase`.

Reply:
I've just merged 310 so that its changes can be incorporated here.